text/0000-devtools-protocol.md

```diff
@@ -24,44 +24,37 @@ See the current protocol here: https://xstate.js.org/docs/packages/xstate-inspec
 `globalThis.__xstate__` is a global `XStateDevInterface` object:
 
 ```ts
-interface InspectedEventObject {
-  name: string; // Event type
-  data: AnyEventObject; // The actual event object
-  origin?: string; // Session ID
-  destination: string; // Session ID
-  createdAt: number; // Timestamp
-}
-
 interface InspectedActorObject {
```

Comment:
Assuming this is the parallel to what we call a Workflow:

- Should we consider adding children to the ActorObject? Workflows have the notion of possible children for self-referential purposes. Do we see this need for InspectedActorObjects?

- Is the purpose of the InspectedActorObject to reflect the status of a running machine? What then, is the purpose of the ActorRef property?

- Why did we change the `events` field to `updates`?

- Workflows are tied explicitly to `MachineVersion` records. Here, we seem to own the `machineDefinition` outright. This may change the way interoperability would work.

Reply:
> Should we consider adding children to the ActorObject? Workflows have the notion of possible children for self-referential purposes. Do we see this need for InspectedActorObjects?

Good idea - will add. Need to consider when there are a _lot_ of actors and it may not be feasible to send them all over - maybe just the IDs?

> Is the purpose of the InspectedActorObject to reflect the status of a running machine? What then, is the purpose of the ActorRef property?

Yes (machine or any other actor logic), but `ActorRef` is only local - it's not sent over the wire.

> Why did we change the events field to updates?

Renamed this event to `ActorTransition`, more accurate

> Workflows are tied explicitly to MachineVersion records. Here, we seem to own the machineDefinition outright. This may change the way interoperability would work.

Changing this to either a JSON-stringified definition _or_ a URL. In Stately's case, this URL would point to the machine resource. This future-proofs us for e.g. self-hosting Stately Studio, local URLs, etc.